app/wizards/placements/add_user_wizard/user_step.rb

```diff
@@ -0,0 +1,29 @@
+class Placements::AddUserWizard::UserStep < Placements::BaseStep
+  attribute :first_name
+  attribute :last_name
+  attribute :email
+
+  validates :first_name, :last_name, presence: true
+  validates :email, presence: true, format: { with: URI::MailTo::EMAIL_REGEXP }
+  validate :new_membership
+
+  delegate :organisation, to: :wizard
+
+  def new_membership
+    return if membership.valid?
+
+    errors.add(:email, :taken)
+  end
+
+  def user
+    @user ||= begin
+      user = Placements::User.find_or_initialize_by(email:)
+      user.assign_attributes(first_name:, last_name:)
+      user
+    end
```

Comment:
I remember you said recently that you try to avoid `begin` blocks where possible.

You might prefer to use `.tap` here.

```suggestion
    @user ||= Placements::User.find_or_initialize_by(email:).tap do |user|
      user.assign_attributes(first_name:, last_name:)
    end
```

_Tomayto, tomahto!_